javascript/src/stores/inputs/InputsStore.ts

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+declare var jsRoutes: any;
+declare var $: any;
+
+import UserNotification = require("../../util/UserNotification");
+import URLUtils = require("../../util/URLUtils");
+
+interface Input {
+    id: string;
+    title: string;
+    description: string;
+    creatorUser: string;
+    createdAt: number;
+}
+
+var InputsStore = {
+    list(callback: ((inputs: Array<Input>) => void)) {
+        var failCallback = (jqXHR, textStatus, errorThrown) => {
+            UserNotification.error("Fetching Inputs failed with status: " + errorThrown,
+                "Could not retrieve Inputs!");
+        };
+
+        $.getJSON(jsRoutes.controllers.api.InputsApiController.list().url, callback).fail(failCallback);
+    },
+    globalRecentMessage(inputId: string, callback: ((message: any) => void)) {
+        var failCallback = (jqXHR, textStatus, errorThrown) => {
+            UserNotification.error("Loading recent message failed with status: " + errorThrown,
+                "Could not retrieve recent message from input " + inputId + "!");
```

Comment:
I think it would help to use the input name to show the error to the user. I think you can accomplish this easily if `javascript/src/components/inputs/InputDropdown.jsx` passes the whole input Object to `javascript/src/components/messageloaders/RecentMessageLoader.jsx`. Then you could pass the name (or the whole input object) to this method and print the nicer error message.